### PR TITLE
Load everything needed for home before we get there.

### PR DESCRIPTION
**Pull request by ArunJ98**

### CODE REVIEW

**Comment by ArunJ98** [comment]
[Move loading of active plan to RedirectScreen](https://trello.com/c/6EjJZWz3/227-move-loading-of-active-plan-to-redirectscreen)

File: frontend/src/home/Home.tsx [hunk 1]
```diff
@@ -457,30 +388,16 @@ class HomeComponent extends React.Component<Props, HomeState> {
...
   logOut = async () => {
     await this.updatePlan(false);
-    this.props.logOut();
     removeAuthTokenFromCookies();
 
     alert(
       "Your plan has been updated and you have been logged out. You will be redirected to the welcome screen."
     );
     this.props.history.push("/");
+    this.props.logOut();
```
Comment (ArunJ98) [inline frontend/src/home/Home.tsx:397]: We need to do this after we dismount Home.

File: frontend/src/Onboarding/RedirectScreen.tsx [hunk 2]
```diff
@@ -99,31 +70,25 @@ export const RedirectScreen: React.FC<Props> = ({ redirectUrl }) => {
               setIsLoading(false); // this update must come last, to make sure other state variables are correctly set before we redirect
             });
           })
-          .catch(e => console.log(e));
+          .catch(e => {
+            // TODO: Log error to some service like rollbar
+            console.log(e);
+            setIsError(true);
```
Comment (NilsBacke) [inline frontend/src/Onboarding/RedirectScreen.tsx:76]: We should also `setIsError(false)` at the start of this function, to handle the case of retrying after there was an error

File: frontend/src/components/TransferCreditHolder.tsx [hunk 3]
```diff
@@ -127,10 +127,10 @@ class TransferCreditsComponent extends React.Component<
     if (this.props.transferCredits == null) {
       return null;
     }
-    return this.props.transferCredits.map((scheduleCourse, index) => {
+    return this.props.transferCredits.map((scheduleCourse, _) => {
```
Comment (NilsBacke) [inline frontend/src/components/TransferCreditHolder.tsx:130]: We can just remove the _ all together

File: frontend/src/home/Home.tsx (hunk 1 above)
Comment (NilsBacke) [reply frontend/src/home/Home.tsx:397]: Just curious, why is this?

**Comment by NilsBacke** [comment]
This looks really really sweet! Also it's nice that we can use the loading screen for anything now. Is the transition between rendering the loading screen in the redirect screen -> home wrapper seemless?

File: frontend/src/home/Home.tsx (hunk 1 above)
Comment (ArunJ98) [reply frontend/src/home/Home.tsx:397]: when we clear stuff from redux, we break the home component

**Comment by ArunJ98** [comment]
> This looks really really sweet! Also it's nice that we can use the loading screen for anything now. Is the transition between rendering the loading screen in the redirect screen -> home wrapper seemless?
Yea, it's pretty seamless, it's just the text changing and the loading indicator might jump a little bit.

**Review by NilsBacke — APPROVED** [review]

**Pull request merged**